core/src/test/java/org/elasticsearch/gateway/MetaDataWriteDataNodesTests.java

```diff
@@ -121,111 +72,22 @@ public void testMetaIsRemovedIfAllShardsFromIndexRemoved() throws Exception {
 
         // create blue_index on blue_node and same for red
         client().admin().cluster().health(clusterHealthRequest().waitForYellowStatus().waitForNodes("3")).get();
```

Comment:
relating the nodes starting above (can't comment on it because it didn't change) - can you use startNodesAsync  and start them together?